### PR TITLE
TST: Fix Unit Tests

### PR DESCRIPTION
**Pull request by maxim-lian**
- Fixed most of these (+ parametrized them). Two remaining are xfailed. I'm not sure whether they should be passing?
- Removed a test testing old dependencies

### CODE REVIEW

**Comment by maxim-lian** [comment]
Gentle ping @tswast @parthea

File: pandas_gbq/tests/test_gbq.py [hunk 1]
```diff
@@ -280,53 +280,31 @@ def test_get_user_account_credentials_returns_credentials(self):
...
-        assert not result
-
-    def test_should_return_bigquery_strings_as_python_strings(self):
-        result = gbq._parse_entry('STRING', 'STRING')
-        assert result == 'STRING'
+    @pytest.mark.parametrize(
+        ('input', 'type_', 'expected'), [
+            (1, 'INTEGER', int(1)),
+            (1, 'FLOAT', float(1)),
+            pytest.param('false', 'BOOLEAN', False, marks=pytest.mark.xfail),
```
Comment (tswast) [inline pandas_gbq/tests/test_gbq.py]: Interesting that this was here before. AFAIK BigQuery returns booleans as proper booleans in the JSON returned. Probably a case of `pandas-gbq` just being extra conservative in what it expects?

File: pandas_gbq/tests/test_gbq.py [hunk 2]
```diff
@@ -280,53 +280,31 @@ def test_get_user_account_credentials_returns_credentials(self):
...
+            pytest.param(
+                '0e9', 'TIMESTAMP',
```
Comment (tswast) [inline pandas_gbq/tests/test_gbq.py]: Does this work if 0 is passed in as a float and not passed as a string?

**Comment by maxim-lian** [comment]
Thanks for merging @tswast  - I was waiting to answer those questions - but you're right better to get it in and come back to them

**Comment by tswast** [comment]
Yeah, I think having unit tests in `master` is more important than solving those right now.

**Pull request merged**